Terraform Docs (#3174)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/functions/getdatasource.mdx

Returns a single data source object that matches the specified type and configuration. The object contains the data source's attributes as returned by the provider, including computed attributes.

### Nested block arguments

> Data source arguments use various types, including nested block arguments. For most argument types, you will assign a value of the matching type within the `config` object. However, when a data source uses nested blocks in Terraform configuration, you must represent them differently in Terraform policy.

Reviewer: ```suggestion
Data source arguments in Terraform configuration can contain one or more nested blocks of arguments. When creating Terraform policy code against data source arguments, you must represent the nested blocks according to the provider's schema.
```
IMO, the parts that aren't about describing nested blocks is confusing.

The type of each argument, including nested blocks, is defined by the provider's schema. There are three types of nested blocks:

> | Block Type | Terraform policy config value |

Reviewer: ```suggestion
You can nest a single block of arguments, a list of arguments, or a set of arguments in Terraform configuration. Assign Terraform policy arguments according to the nested block's data type:

| Block type | Terraform policy config value |
```
When I first started learning TF language, terminology was a major barrier. Things became easier after I de-conceptualized some of the language. For example, I eventually realized that "meta-arguments" just means something like "built-in arguments" as opposed to provider arguments or that "policy set" literally just means "a set of policies". It's starting to occur to me that perhaps "nested block" is also potentially over-conceptualized and that instead of talking about "nested blocks" we should talk about "a block of arguments nested in an argument" . . . or something like that. Feel free to disagree. Apologies for the long comment.

Author: So you're nesting blocks - either single, list, or set, and each of those blocks has its own arguments. I think it would be correct to say:

```suggestion
You can nest a single block of arguments, a list of blocks of arguments, or a set of blocks of arguments in Terraform configuration. Assign Terraform policy arguments according to the nested block's data type::

| Block type | Terraform policy config value |
```

...but that seems overall less clear than the original to me.

Reviewer: ```suggestion
In Terraform configuration, you can nest a single block, a list of blocks, or a set of blocks. Each block can contain its own arguments. Assign Terraform policy arguments according to the nested block's data type:
```

|------------|-------------------------------|
| Single nested | Assign an object value to an attribute with the same name as the block. |
| List nested | Assign a list of object values. |
| Set nested  | Assign a list of object values. |

> For example, the AWS `aws_ami` data source uses a set nested block for `filter`. In Terraform configuration, you would write:

Reviewer: ```suggestion
For example, the AWS `aws_ami` data source lets you define a set of nested blocks to filter responses.  In Terraform configuration, you could write the following `filter` blocks to filter AMIs by name and virtualization type:
```
